### DISCUSSION TITLE
setMatrix后，pag的预设效果丢失

### DISCUSSION THREAD

**Discussion by l540144279** [Tencent/libpag#1725] [Q&A] [answered]
预设了pag的背景的高斯模糊效果，填充图片后等比拉伸填满长边，其余部分会显示该图片的高斯模糊效果，此时对图片的matrix做scale操作，repalceImage后高斯模糊效果丢失。设置代码如下：

`auto matrix = image->matrix();
  matrix.postScale(scale, scale, 0, 0);
  image->setMatrix(matrix);
  pagFile->replaceImage(slot, image);
`
[合成 1-0.pag.zip](https://github.com/Tencent/libpag/files/12575741/1-0.pag.zip)

**Comment by domchen** [comment, marked as answer]
应该不是高斯模糊消失了，而是被你挡住了。你这个pag是一张图片同时在两个图层里引用，但是有不同的缩放值。你如果在同一个PAGImage上设置matrix，会同时影响两个引用它的图层，都变成一样大了就完全挡住了后面那个。你应该是获取对应的PAGImageLayer，修改单个PAGImageLayer的matrix来缩放。或者干脆做成两个图片分别替换。

**Comment by Jasper2016** [comment]
为什么底图用高斯模糊后会有透明边框，底图已经放大，远超过合成尺寸。底图已经是zoom的填充方式，背景图和内容图是同一张图片但是换了名字。
<img width="672" alt="WechatIMG78" src="https://github.com/Tencent/libpag/assets/20656580/f28a4b25-d35a-45cd-948c-2a103dd2ec3e">
